annotations/views/annotation_views.py

```diff
@@ -35,14 +36,22 @@ def annotate_image(request, text_id):
 def relations(request):
     from annotations.filters import RelationSetFilter
 
-    qs = RelationSet.objects.all()
-    filtered = RelationSetFilter(request.GET, queryset=qs)
+
+    filtered = RelationSetFilter(request.GET, queryset=RelationSet.objects.all())
     qs = filtered.qs
     for r in qs:
         print r.__dict__
 
     paginator = Paginator(qs, 40)
     page = request.GET.get('page')
+
+    project = filtered.data.get('project')
+    createdBy = filtered.data.get('createdBy')
+    occursIn = filtered.data.get('occursIn')
+    createdAfter = filtered.data.get('createdAfter')
+    createdBefore = filtered.data.get('createdBefore')
+    terminal_nodes = filtered.data.get('terminal_nodes')
```

Comment:
I know you don't want to hear that, but this way your data is not yet validated. Instead, call ```is_valid()``` on the form in the filter and then put ```cleaned_data``` into your context. In your template, simply iterate over cleaned_data and append key=value to the url. This way if the filter changes, the rest of the code doesn't have to be changed accordingly -> less maintenance -> fewer possibilities for bugs.